File: src/main/java/com/chatchatabc/jpademojava/domain/service/CountryService.java
package com.chatchatabc.jpademojava.domain.service;

import com.chatchatabc.jpademojava.domain.model.Country;
import com.chatchatabc.jpademojava.domain.model.User;
import org.springframework.stereotype.Service;

@Service
public interface CountryService {

    /**
     * Create country
     *
     * @param country
     * @return
     */
    Country create(Country country);

    /**
     * Update country
     *
     * @param countryId
     * @param newCountryInfo
     * @return
     */
    Country update(String countryId, Country newCountryInfo) throws Exception;

    /**
     * Assign user to country
     *
     * @param userId
     * @param countryId
     * @return
     */
    User assign(String userId, String countryId) throws Exception;

    /**
     * Unassign user from country
     *
     * @param userId
     * @return
     * @throws Exception
     */
    User unassign(String userId) throws Exception;
}

File: src/main/kotlin/com/chatchatabc/jpademo/domain/service/CountryService.kt
package com.chatchatabc.jpademo.domain.service

import com.chatchatabc.jpademo.domain.model.Country
import com.chatchatabc.jpademo.domain.model.User
import org.springframework.stereotype.Service

@Service
interface CountryService {
    /**
     * Create Country
     */
    fun create(country: Country): Country

    /**
     * Update Country
     */
    fun update(countryId: String, newCountryInfo: Country): Country

    /**
     * Assign Country to User
     */
    fun assign(userId: String, countryId: String): User

    /**
     * Unassign Country from User
     */
    fun unassign(userId: String): User
}